providers/core.tcpsocket.js

```diff
@@ -39,7 +39,17 @@ Socket_chrome.prototype.getInfo = function(continuation) {
   if (this.id) {
     // Note: this.namespace used, since this method is common to tcp and
     // tcpServer sockets.
-    this.namespace.getInfo(this.id, continuation);
+    this.namespace.getInfo(this.id, function(info) {
+      // This can happen if the socket is disconnected
+      // before the callback is invoked.
+      if (chrome.runtime.lastError) {
```

Comment:
based on how I read the description of `lastError` https://developer.chrome.com/extensions/runtime#property-lastError , it will be set explicity for the getInfo failing, and that error will be lost if we don't do something with it in that callback. Is there a reason not to call `dispatchDisconnect()` at this point, so that the error that resulted in the socket now being closed is relayed to the client?